Add zoo-info-viewer --print-acls usage, document expected ZooKeeper ACLs (#371)
apache/accumulo-website: _docs-2/troubleshooting/zookeeper.md

category: troubleshooting
order: 7
---
## ZooKeeper ACLs

Accumulo requires full access to nodes in ZooKeeper under the /accumulo path.  The ACLs can be examined using the
ZooKeeper cli `getAcl` and modified with `setAcl` commands.  With 2.1.1, the zoo-info-viewer utility has an option

> that will print all of the ACLs for the nodes under `/accumulo/[INSTANCE_ID]` (See the [zoo-info-viewer]).  

Reviewer: ```suggestion
that will print all of the ACLs for the nodes under `/accumulo/[INSTANCE_ID]` (See [zoo-info-viewer]).  
```

To run the utility, only ZooKeeper needs to be running. If hdfs is running, the instance id can be read from hdfs, 
or it can be entered with the zoo-info-viewer command --instanceId option.  Accumulo management processes 
*do not* need to be running. This allows checking the ACLs before starting an upgrade. 

> 

Reviewer: Might be useful to include a one or two line example output.

Author: Added in 12938808

> The utility prints out two summarized fields related to ZooKeeper ACL permissions:

Reviewer: ```suggestion
The utility prints out a line for each znode that contains two fields related to ZooKeeper ACL permissions:
```

   - `[ACCUMULO_OKAY | ERROR_ACCUMULO_MISSING_SOME]` - Are the permissions sufficient for Accumulo to operate 
   - `[PRIVATE | NOT_PRIVATE]` - Can other users can read data from the ZooKeeper nodes.

> If there are nodes with `ERROR_ACCUMULO_MISSING_SOME` are nodes where Accumulo does not have `cdrwa` permissions.

Reviewer: ```suggestion
Nodes marked with `ERROR_ACCUMULO_MISSING_SOME` means that Accumulo does not have `cdrwa` permissions.
```
